src/nvim/eval.c

```diff
@@ -12620,6 +12630,44 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
         return;
       }
     }
+    dictitem_T *item;
+    item = tv_dict_find(job_opts, S_LEN("env"));
+    if (item) {
+      size_t custom_env_size = (size_t)tv_dict_len(item->di_tv.vval.v_dict);
+      size_t i = 0;
+      size_t env_size = 0;
+      if (item->di_tv.v_type != VAR_DICT) {
+        EMSG2(_(e_invarg2), "env");
```

Comment:
missing test

Reply:
You mean you don't like crashes when the type is wrong? :)